device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java

```diff
@@ -138,7 +161,7 @@
 
     public void open() throws IOException
     {
-        if (this.config.getAuthenticationType() == DeviceClientConfig.AuthType.SAS_TOKEN && this.config.getSasTokenAuthentication().isRenewalNecessary())
+        if (this.config.getAuthenticationType() == DeviceClientConfig.AuthType.SAS_TOKEN && this.config.getSasTokenAuthentication().isAuthenticationProviderRenewalNecessary())
```

Comment:
nit: I am a bit confused by the naming of this - we aren't renewing the authentication provider itself, right?

Reply:
We kind of are. ```isAuthenticationProviderRenewalNecessary``` is only true when the user provides a single SAS token, and it expires, and we can't refresh it because we don't have the device key

Reply:
So are they really renewing the authentication provider or just a SAS token? the name is really confusing to me.

Reply:
so does the SasTokenProvider control the auth keys to generate the SAS token?

Reply:
I'd just ignore this function, to be honest. It only is relevant in very niche cases. For users who use the new DeviceClient constructor in this PR, they will never enter into this if block.